docs/explanations/decisions/0003-make-devices-factory.md

```diff
@@ -0,0 +1,28 @@
+# 3. Add device factory decorator with lazy connect support
+
+Date: 2024-04-26
+
+## Status
+
+Accepted
+
+## Context
+
+We should add a decorator to support verfied device connection.
+
+## Decision
+
+DAQ members led us to this proposal:
+
+- ophyd-async: make Device.connect(mock, timeout, force=False) be idempotent
+- ophyd-async: make ensure_connected(\*devices) plan stub
+- dodal: make device_factory(startup_connect=False) decorator that makes, names, caches and optionally connects a device
```

Comment:
```suggestion
- dodal: make device_factory(eager=False) decorator that makes, names, caches and optionally connects a device
```

names of the decorator params have changed, may be other changes in here to reflect

Reply:
the ADR reflected the understanding in April, it's allowed to change. let's not change records of the past too easily :eye: ,  ADR is not a docstring

Reply:
It's a new file though. I could maybe understand that perspective if it were already in the codebase, but since it isn't everything in it should be subject to change to make it more accurate to the change its referencing.